src/ProductsContext/ProductState.tsx

```diff
@@ -28,7 +28,27 @@ const ProductState: React.FC<ProductStateProps> = (props: any) => {
     const json = await response.json();
     return json
   }
-  const value: any = { getProducts, products, setName, setPassword, name, password, handleLogin };
+
+  const handleGetDetails = async (id: number) => {
+    const check = await fetch('https://dummyjson.com/auth/me', {
```

Comment:
you will be using this logic with every api. so why not move this auth/me into a separate fn?
that way, for every api call you do in service files, you presumably start with this step.

make sure all apis after login are using this!